src/ucp/core/ucp_ep.c

```diff
@@ -447,6 +468,13 @@ static void ucp_ep_disconnected(ucp_request_t *req)
 {
     ucp_ep_h ep = req->send.ep;
 
+    /* Delay cleanup of failed EPs until worker destroy */
+    if (ucp_ep_config(ep)->status != UCS_OK) {
```

Comment:
would be better to check ep->flags here, if needed define more flags for err handling